src/carnot/planner/compiler/ast_visitor.cc

```diff
@@ -921,14 +921,20 @@ StatusOr<QLObjectPtr> ASTVisitorImpl::ProcessTuple(const pypa::AstTuplePtr& ast,
   return TupleObject::Create(ast, expr_vec, this);
 }
 
-StatusOr<QLObjectPtr> ASTVisitorImpl::ProcessNumber(const pypa::AstNumberPtr& node) {
+StatusOr<QLObjectPtr> ASTVisitorImpl::ProcessNumber(const pypa::AstNumberPtr& node, bool negate) {
   switch (node->num_type) {
     case pypa::AstNumber::Type::Float: {
+      if (negate) {
+        node->floating *= -1;
+      }
       PX_ASSIGN_OR_RETURN(FloatIR * ir_node, ir_graph_->CreateNode<FloatIR>(node, node->floating));
       return ExprObject::Create(ir_node, this);
     }
     case pypa::AstNumber::Type::Integer:
     case pypa::AstNumber::Type::Long: {
```

Comment:
I thought `pypa::AstNumber::Type::Long` types would need to be handled differently (as they are in [libpypa's parser](https://github.com/pixie-io/libpypa/blob/pixie/src/pypa/parser/parser.cc#L1164)), but from the unit tests I added that doesn't seem to be the case.

Adding this reference here since I'm new to this code and that surprised me.